packages/datadog-plugin-aws-sdk/src/services/kinesis.js

```diff
@@ -1,15 +1,64 @@
 'use strict'
-
+const log = require('../../../dd-trace/src/log')
 class Kinesis {
   generateTags (params, operation, response) {
-    const tags = {}
-
-    if (!params || !params.StreamName) return tags
+    if (!params || !params.StreamName) return {}
 
-    return Object.assign(tags, {
+    return {
       'resource.name': `${operation} ${params.StreamName}`,
       'aws.kinesis.stream_name': params.StreamName
-    })
+    }
+  }
+
+  // AWS-SDK will b64 kinesis payloads
+  // or will accept an already b64 encoded payload
+  // This method handles both
+  _tryParse (body) {
+    try {
+      return JSON.parse(body)
+    } catch (e) {
+      log.info('Not JSON string. Trying Base64 encoded JSON string')
+    }
+    try {
+      return JSON.parse(Buffer.from(body, 'base64').toString('ascii'), true)
+    } catch (e) {
+      return null
```

Comment:
The error should be logged.

Reply:
It's logged by the calling function, but I can add another log line here

Reply:
For the error to be logged by the caller it would have to not be caught here.

Reply:
That's why the null check catches it https://github.com/DataDog/dd-trace-js/pull/1776/files#diff-fff221624b62118bef1afebd04ea2f1c257d823903d666f808c48c2eb93e373bR59